src/screens/Bookmarks/Bookmarks.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useIsFocused } from '@react-navigation/native';
+import { Text, View, SafeAreaView, ScrollView } from 'react-native';
+import { BookmarksStackScreenProps } from '../../types/navigation';
+import JobCard from '../../components/JobCard/JobCard';
+import { Job } from '../../types/types';
+import styles from './styles';
+import {
+  getAllBookmarks,
+  updateUserBookmarks,
+} from '../../firebase/firestore/user';
+import { AuthContext } from '../../context/AuthContext';
+
+function BookmarksScreen({
+  navigation,
+}: BookmarksStackScreenProps<'BookmarksScreen'>) {
+  const [bookmarkedList, setBookmarkedList] = useState([] as Job[]);
+  const { userObject } = useContext(AuthContext);
+  const userBookmarkedJobs = userObject?.bookmarkedJobs;
+
+  const isFocused = useIsFocused();
+
+  useEffect(() => {
+    const fetchJobs = async () => {
+      const data = await getAllBookmarks(userObject);
+      setBookmarkedList(data);
+    };
+    fetchJobs();
+  }, [isFocused, userObject?.id, userObject]);
+
+  useEffect(() => {
+    const unsubscribe = navigation.addListener('blur', async () => {
+      if (userObject?.id === undefined) {
+        return;
+      }
+      await updateUserBookmarks(userBookmarkedJobs, userObject?.id);
```

Comment:
i think if you follow what the comment i wrote about the getBookmarks function then i think you actually don't need any arguments to this function and can just use the userObject you get in the user.ts file